src/NHibernate/Dialect/SQLiteDialect.cs

```diff
@@ -513,5 +518,72 @@ protected override bool CastingIsRequired(string sqlType)
 				return true;
 			}
 		}
+
+		[Serializable]
+		private class LocateFunction : ISQLFunction, ISQLFunctionExtended
+		{
+			// Since v5.3
+			[Obsolete("Use GetReturnType method instead.")]
+			public IType ReturnType(IType columnType, IMapping mapping)
+			{
+				return NHibernateUtil.Int32;
+			}
+
+			/// <inheritdoc />
+			public IType GetReturnType(IEnumerable<IType> argumentTypes, IMapping mapping, bool throwOnError)
+			{
+#pragma warning disable 618
+				return ReturnType(argumentTypes.FirstOrDefault(), mapping);
+#pragma warning restore 618
+			}
+
+			/// <inheritdoc />
+			public IType GetEffectiveReturnType(IEnumerable<IType> argumentTypes, IMapping mapping, bool throwOnError)
+			{
+				return GetReturnType(argumentTypes, mapping, throwOnError);
+			}
+
+			/// <inheritdoc />
+			public string Name => "instr";
+
+			public bool HasArguments => true;
+
+			public bool HasParenthesesIfNoArguments => true;
+
+			public SqlString Render(IList args, ISessionFactoryImplementor factory)
+			{
+				if (args.Count != 2 && args.Count != 3)
+				{
+					throw new QueryException("'locate' function takes 2 or 3 arguments. Provided count: " + args.Count);
+				}
+
+				if (args.Count == 2)
+				{
+					return new SqlString("instr(", args[1], ", ", args[0], ")");
+				}
+
+				var text = args[1];
+				var value = args[0];
+				var startIndex = args[2];
+				//ifnull(
```

Comment:
Other nit: I tend to favor use of functions which are in the SQL Standard, which is the case of `coalesce`, supported by SQLite, and which can replace `ifNull` without changing anything more that the function name. I do not know with certainty if `ifNull` is in the standard, but I think it is not.
(I have not found any reference about it being in the standard or not. It is supported by MySql, SQLite, Big Query, but not by SQL-Server (which has `isNull`) nor Oracle (`nvl`), while they all support `coalesce`.)

Reply:
This is a dialect specific function implementation, so using non-standard function should be fine here.